src/core/tools/readFileTool.ts

```diff
@@ -52,69 +47,43 @@ export async function readFileTool(
 			const { maxReadFileLine = 500 } = (await cline.providerRef.deref()?.getState()) ?? {}
 			const isFullRead = maxReadFileLine === -1
 
-			// Check if we're doing a line range read
-			let isRangeRead = false
-			let startLine: number | undefined = undefined
-			let endLine: number | undefined = undefined
-
-			// Check if we have either range parameter and we're not doing a full read
-			if (!isFullRead && (startLineStr || endLineStr)) {
-				isRangeRead = true
-			}
-
-			// Parse start_line if provided
+			// Parse start_line if provided (keep as 1-based)
+			let requestedStartLine: number | undefined = undefined
 			if (startLineStr) {
-				startLine = parseInt(startLineStr)
-
-				if (isNaN(startLine)) {
-					// Invalid start_line
+				requestedStartLine = parseInt(startLineStr)
+				if (isNaN(requestedStartLine)) {
 					cline.consecutiveMistakeCount++
 					cline.recordToolError("read_file")
 					await cline.say("error", `Failed to parse start_line: ${startLineStr}`)
 					pushToolResult(`<file><path>${relPath}</path><error>Invalid start_line value</error></file>`)
 					return
 				}
-
-				startLine -= 1 // Convert to 0-based index
 			}
 
-			// Parse end_line if provided
+			// Parse end_line if provided (keep as 1-based)
+			let requestedEndLine: number | undefined = undefined
 			if (endLineStr) {
-				endLine = parseInt(endLineStr)
-
-				if (isNaN(endLine)) {
-					// Invalid end_line
+				requestedEndLine = parseInt(endLineStr)
+				if (isNaN(requestedEndLine)) {
 					cline.consecutiveMistakeCount++
 					cline.recordToolError("read_file")
 					await cline.say("error", `Failed to parse end_line: ${endLineStr}`)
 					pushToolResult(`<file><path>${relPath}</path><error>Invalid end_line value</error></file>`)
 					return
 				}
-
-				// Convert to 0-based index
-				endLine -= 1
-			}
-
-			const accessAllowed = cline.rooIgnoreController?.validateAccess(relPath)
-
-			if (!accessAllowed) {
-				await cline.say("rooignore_error", relPath)
-				const errorMsg = formatResponse.rooIgnoreError(relPath)
-				pushToolResult(`<file><path>${relPath}</path><error>${errorMsg}</error></file>`)
-				return
 			}
 
 			// Create line snippet description for approval message
 			let lineSnippet = ""
 
 			if (isFullRead) {
 				// No snippet for full read
-			} else if (startLine !== undefined && endLine !== undefined) {
-				lineSnippet = t("tools:readFile.linesRange", { start: startLine + 1, end: endLine + 1 })
-			} else if (startLine !== undefined) {
-				lineSnippet = t("tools:readFile.linesFromToEnd", { start: startLine + 1 })
-			} else if (endLine !== undefined) {
-				lineSnippet = t("tools:readFile.linesFromStartTo", { end: endLine + 1 })
+			} else if (requestedStartLine !== undefined && requestedEndLine !== undefined) {
```

Comment:
These look like unnecessary variable rename that cause diff review bloat. 

Unless there is a really good reason to rename variables let's not do that if we can avoid it, so we can tell what actually change for the pull request, and what is original .